libraries/core-react/src/Tabs/TabList.tsx

```diff
@@ -1,38 +1,61 @@
-// @ts-nocheck
 import React, {
   forwardRef,
   useContext,
   useRef,
   useCallback,
   useEffect,
+  ReactElement,
 } from 'react'
-import PropTypes from 'prop-types'
 import styled from 'styled-components'
 import { useCombinedRefs } from '../_common/useCombinedRefs'
 import { TabsContext } from './Tabs.context'
+import { Variants } from './Tabs.types'
 
-const variants = {
+type VariantsRecord = {
+  fullWidth: string
+  minWidth: string
+}
+
+const variants: VariantsRecord = {
   fullWidth: 'minmax(1%, 360px)',
   minWidth: 'max-content',
 }
 
-const StyledTabList = styled.div.attrs(() => ({
-  role: 'tablist',
-}))`
+type StyledProps = Props
+
+const StyledTabList = styled.div.attrs(
+  (): React.HTMLAttributes<HTMLDivElement> => ({
+    role: 'tablist',
+  }),
+)<StyledProps>`
   display: grid;
   grid-auto-flow: column;
   grid-auto-columns: ${({ variant }) => variants[variant]};
 `
 
-const TabList = forwardRef(function TabsList({ children, ...props }, ref) {
-  const { activeTab, handleChange, tabsId, variant, tabsFocused } = useContext(
-    TabsContext,
-  )
+type Props = {
+  /** Sets the width of the tabs */
+  variant?: Variants
+} & React.HTMLAttributes<HTMLDivElement>
```

Comment:
`children` should be mentioned here as it is required, right? 🤔

Reply:
As discussed during standup. Added tests and fallbacks for if children is not defined to stay as close as possible to native behaviour